src/nvhttp.cpp

```diff
@@ -267,12 +268,32 @@ namespace nvhttp {
 
     launch_session.host_audio = host_audio;
     launch_session.gcm_key = util::from_hex<crypto::aes_t>(get_arg(args, "rikey"), true);
+    std::stringstream mode = std::stringstream(get_arg(args, "mode"));
+    // Split mode by the char "x", to populate width/height/fps
+    int x = 0;
+    std::string segment;
+    while (std::getline(mode, segment, 'x')) {
+      if (x == 0) launch_session.width = atoi(segment.c_str());
+      if (x == 1) launch_session.height = atoi(segment.c_str());
+      if (x == 2) launch_session.fps = atoi(segment.c_str());
+      x++;
+    }
+    launch_session.unique_id = (get_arg(args, "uniqueid"));
+    launch_session.uuid = (get_arg(args, "uuid"));
+    launch_session.appid = util::from_view(get_arg(args, "appid"));
+    launch_session.enable_sops = util::from_view(get_arg(args, "sops"));
+    launch_session.surround_info = util::from_view(get_arg(args, "surroundAudioInfo"));
+    launch_session.gcmap = util::from_view(get_arg(args, "gcmap"));
+    launch_session.enable_hdr = 0;
+    if (args.find("enableHdr"s) != std::end(args)) {
+      launch_session.enable_hdr = util::from_view(get_arg(args, "enableHdr"));
```

Comment:
@TheElixZammuto 

Can we set a default value to empty/null if the client doesn't provide it? I'm thinking something similar to Python's `os.getenv` method. `os.getenv(key, "some default value")`

Apparently, Moonlight on Apple doesn't provide UUID (and who knows about every possible homebrew moonlight client). https://discord.com/channels/804382334370578482/1074377411069747250/1135639228613664858

Reply:
Perhaps something like

```cpp
  std::string
  get_arg(const args_t &args, const char *name, const char *default_value = nullptr) {
    auto it = args.find(name);
    if (it == std::end(args)) {
      if (default_value != NULL) {
        return std::string(default_value);
      }
      
      throw std::out_of_range(name);
    }
    return it->second;
  }
```

Used as such:
```cpp
   launch_session.gcm_key = util::from_hex<crypto::aes_t>(get_arg(args, "rikey", "0"), true);
    std::stringstream mode = std::stringstream(get_arg(args, "mode", "0x0x0"));
    // Split mode by the char "x", to populate width/height/fps
    int x = 0;
    std::string segment;
    while (std::getline(mode, segment, 'x')) {
      if (x == 0) launch_session.width = atoi(segment.c_str());
      if (x == 1) launch_session.height = atoi(segment.c_str());
      if (x == 2) launch_session.fps = atoi(segment.c_str());
      x++;
    }
    
    launch_session.unique_id = (get_arg(args, "uniqueid", "0123456789ABCDEF"));
    launch_session.uuid = (get_arg(args, "uuid", "00000000000000000000000000000000"));
    launch_session.appid = util::from_view(get_arg(args, "appid", "0123456789"));
    launch_session.enable_sops = util::from_view(get_arg(args, "sops", "0"));
    launch_session.surround_info = util::from_view(get_arg(args, "surroundAudioInfo", "0"));
    launch_session.gcmap = util::from_view(get_arg(args, "gcmap", "0"));
    launch_session.enable_hdr = 0;
    if (args.find("enableHdr"s) != std::end(args)) {
      launch_session.enable_hdr = util::from_view(get_arg(args, "enableHdr", "0"));
    }
    uint32_t prepend_iv = util::endian::big<uint32_t>(util::from_view(get_arg(args, "rikeyid", "0")));
```